src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -265,6 +265,39 @@ private static string EncodeName(string name)
             return name;
         }
 
+        private static string EncodePropertyName(string name)
+        {
+            // convert C# indexer names to 'Item'
+            if (name == "this[]")
+            {
+                name = "Item";
+            }
+            else if (name.EndsWith(".this[]"))
+            {
+                name = name.Substring(0, name.Length - 6) + "Item";
```

Comment:
Instead of `6`, I often write "this[]".Length in cases like these, because I'm bad at counting.